### ISSUE TITLE
Published sourcemaps contain absolute paths in publisher's home directory

### ISSUE THREAD

**Issue by jsnajdr** [WordPress/gutenberg#13883]
1. Install the `@wordpress/api-fetch` package into your project.
2. Look at the `node_modules/@wordpress/api-fetch/build/index.js.map` file.

The `sources` field near the beginning contains an absolute path into Riads home directory:
```
/Users/riad/Workspace/a8c/gutenberg/packages/api-fetch/src/index.js
```

I discovered this when upgrading Calypso to Jest 24.1 in Automattic/wp-calypso#30434. Jest uses sourcemaps to show error stack traces, which are therefore confusing:
```
    ReferenceError: regeneratorRuntime is not defined

      at ../../../riad/Workspace/a8c/gutenberg/packages/api-fetch/src/middlewares/fetch-all-middleware.js:42:25
      at Object.<anonymous> (../../../riad/Workspace/a8c/gutenberg/packages/api-fetch/src/middlewares/fetch-all-middleware.js:42:25)
      at Object.<anonymous> (../../../riad/Workspace/a8c/gutenberg/packages/api-fetch/src/index.js:12:1)
      at Object.<anonymous> (client/gutenberg/editor/api-middleware/index.js:9:1)
      at Object.<anonymous> (client/gutenberg/editor/api-middleware/test/index.js:9:1)
```

(don't pay much attention to the `regeneratorRuntime` error, it's not related to this bug)

**Comment by earnjam** [comment]
I brought this up in Slack a couple of months ago.
https://wordpress.slack.com/archives/C5UNMSU4R/p1540392138000100

Don’t know that we ever made a ticket for it though.

**Comment by jsnajdr** [comment]
I don't know what the right solution is. Some projects use `webpack:` pseudo-URLs, others ship relative paths: `../src/index.js`. That's what RxJS does, for example. But they ship the untranspiled sources in `src`, too, unlike Gutenberg, so the file references are valid.

At least the file paths in the error stacktrace would make more sense:
```
node_modules/@wordpress/api-fetch/src/middlewares/fetch-all-middleware.js
```

**Comment by gziolo** [comment]
Fix proposed in #14409.
